deploy/helm/scf/assets/operations/instance_groups/api.yaml

```diff
@@ -234,31 +307,56 @@
     run:
       healthcheck:
         policy-server:
-          readiness: &policy_server_readiness
+          readiness:
+            httpGet:
+              port: 4002
+          liveness:
             exec:
-              command:
-              - sh
-              - -c
-              - ss -nlt | grep "LISTEN.*:4002"
+              command: [pidof, policy-server]
     post_start:
-      condition: *policy_server_readiness
+      condition:
+        exec:
+          # policy-server doesn't support HTTP HEAD requests
+          command: [curl, --fail, --silent, http://localhost:4002/]
 
 # Add quarks properties for policy-server-internal.
 - type: replace
   path: /instance_groups/name=api/jobs/name=policy-server-internal/properties/quarks?
   value:
     run:
       healthcheck:
-        policy-server:
+        policy-server-internal:
           readiness: &policy_server_internal_readiness
+            httpGet:
+              port: 31946
+          liveness:
             exec:
-              command:
-              - sh
-              - -c
-              # TODO: Use curl to call port 31946.
-              - ss -nlt | grep "LISTEN.*:4003"
+              command: [pidof, policy-server-internal]
     post_start:
-      condition: *policy_server_internal_readiness
+      condition:
+        exec:
+          # policy-server-internal doesn't support HTTP HEAD requests
+          command: [curl, --fail, --silent, http://localhost:31946/]
+
+- type: replace
+  path: /instance_groups/name=api/jobs/name=route_registrar/properties/quarks?/run/healthcheck/route_registrar
+  value:
+    readiness: &route_registrar_readiness
+      # The route registrar doesn't expose anything to indicate if the
```

Comment:
You don't have to do that. The container daemon does it for you.
If the process goes away, the container should go away.

Reply:
Our process tree is
```
/usr/bin/dumb-init
\_ /var/vcap/all-releases/container-run/container-run
  \_ /var/vcap/packages/route_registrar/bin/route-registrar
```
(Or similar for the other things)

So we could potentially get stuck and never run the route-registrar process / not notice it exiting.  If we had `route-registrar` as pid 1, then the `pidof` checks would definitely be unnecessary.

I'm okay with just getting rid of the `pidof` checks completely, though; I feel like that's less crazy then just having the same liveness & readiness probes.

Reply:
If we do the `pidof` approach, we are saying that we don't trust the software we wrote (ok, I did) and has 100% test coverage. Unless we also don't trust dumb-init.

Reply:
Already got rid of `pidof` :) I blame the task for being specified as add liveness & readiness probes, and not _as appropriate_ :stuck_out_tongue: So I added a bunch inappropriately :grin: